posthog/api/person.py

```diff
@@ -353,18 +358,29 @@ def calculate_path_persons(self, request: request.Request) -> Dict[str, Tuple[li
 
     @action(methods=["GET"], detail=False)
     def values(self, request: request.Request, **kwargs) -> response.Response:
-        people = self.get_queryset()
-        key = "properties__{}".format(request.GET.get("key"))
-        people = people.values(key).annotate(count=Count("id")).filter(**{f"{key}__isnull": False}).order_by("-count")
-
-        if request.GET.get("value"):
-            people = people.extra(
-                where=["properties ->> %s LIKE %s"], params=[request.GET["key"], "%{}%".format(request.GET["value"])],
+        timer = statsd.timer("person_property_values_timer").start()
```

Comment:
Instead of having 2 nested `try/except`, what do you think about something like this? 

```python
@action(methods=["GET"], detail=False)
def values(self, request: request.Request, **kwargs) -> response.Response:
    key = request.GET.get("key")
    value = request.GET.get("value")
    
    flattened = []

    if key:
        timer = statsd.timer("get_person_property_values_for_key_timer").start()
        try:
            result = get_person_property_values_for_key(key, self.team, value)
        except Exception as e:
            statsd.incr(
                "get_person_property_values_for_key_failure",
                tags={"error": str(e), "key": key, "value": value, "team_id": self.team.id},
            )
            raise e
        finally:
            timer.stop()

        statsd.incr(
            "get_person_property_values_for_key_success"
            tags={"key": key, "value": value, "team_id": self.team.id},
        )        
        for (value, count) in result:
            try:
                # Try loading as json for dicts or arrays
                flattened.append({"name": convert_property_value(json.loads(value)), "count": count})  # type: ignore
            except json.decoder.JSONDecodeError:
                flattened.append({"name": convert_property_value(value), "count": count})
    
    return response.Response(flattened)
```
Note: the behaviour should be the same but I didn't test it

Reply:
I guess this has the (maybe small) benefit that the timer isn't including the time to loop over the values and convert them to JSON

Feels bad that we need to write this code everywhere instead of something like `@timed('get_person_property_values_for_key_timer')`

Reply:
updated